auth/src/main/java/org/aerogear/android/ags/auth/credentials/OIDCCredentials.java

```diff
@@ -0,0 +1,206 @@
+
+        try {
+            //  Validate the JWT and process it to the Claims
+            JwtClaims jwtClaims = jwtConsumer.processToClaims(jwtToken);
+            return true;
+        } catch (InvalidJwtException e) {
+            Log.e(TAG, "Invalid JWT provided", e);
+            return false;
+        }
+    }
+
+    /**
+     * Returns whether this token is expired or not.
+     * @return <code>true</code> if expired.
+     */
+    public boolean isExpired() {
+        return authState.hasClientSecretExpired();
+    }
+
+    /**
+     * Check whether new access token is needed.
+     * @return <code>true</code> if access token is needed
+     */
+    public boolean getNeedsRenewal() {
+        return authState.getNeedsTokenRefresh();
+    }
+
+    /**
+     * Force request of new access token.
+     */
+    public void setNeedsRenewal() {
+        authState.setNeedsTokenRefresh(true);
+    }
+
+    /**
+     * Check if the user is authenticated/authorized.
+     * @return <code>true</code> if the user is authenticated/authorized.
+     */
+    public boolean isAuthorized() {
+        return authState.isAuthorized();
+    }
+
+    /**
+     * Returns stringified JSON for the OIDCCredential.
+     * @return Stringified JSON OIDCCredential
+     */
+    public String serialize() {
+        try {
+            final JSONObject jsonCredential = new JSONObject()
+                .put("authState", this.authState.jsonSerializeString());
+            if (this.integrityCheckParameters != null) {
+                jsonCredential.put("integrityCheck", this.integrityCheckParameters.serialize());
+            }
+            return jsonCredential.toString();
+        } catch(JSONException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    public static OIDCCredentials deserialize(final String serializedCredential) {
+        try {
+            final JSONObject jsonCredential = new JSONObject(serializedCredential);
+            final String serializedAuthState = jsonCredential.getString("authState");
+            final String serializedIntegrityChecks = jsonCredential.getString("integrityCheck");
+            final IntegrityCheckParameters icParams = IntegrityCheckParameters.deserialize(serializedIntegrityChecks);
+            return new OIDCCredentials(serializedAuthState, icParams);
+        } catch(JSONException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    /**
+     * Check whether the user is authorized and not expired.
+     * @return <code>true</code> if user is authorized and token is not expired.
+     */
+    public boolean checkValidAuth() {
+        return isAuthorized() && getNeedsRenewal();
+    }
+
+    /**
+     * Renew the token
+     * @return
+     * @throws AuthenticationException
+     */
+    public boolean renew() throws AuthenticationException {
+        throw new IllegalStateException("Not yet implemented");
```

Comment:
Need implementation :)

Reply:
Not sure why this is showing up as changed, but the renew was there before this PR and I think it's outside the scope of this PR. I think its implementation is kind of dependent on knowing how login will work.